protostar/commands/format/format_command.py

```diff
@@ -0,0 +1,119 @@
+from logging import Logger
+from typing import List, Optional, Tuple
+from pathlib import Path
+
+from starkware.cairo.lang.compiler.parser import parse_file
+from starkware.cairo.lang.compiler.parser_transformer import ParserError
+
+from protostar.cli import Command, ActivityIndicator
+from protostar.protostar_exception import ProtostarExceptionSilent
+from protostar.utils import log_color_provider
+
+from protostar.commands.format.formatting_summary import FormatingSummary
+from protostar.commands.format.formatting_result import (
+    BrokenFormattingResult,
+    CorrectFormattingResult,
+    IncorrectFormattingResult,
+)
+
+
+class FormatCommand(Command):
+    def __init__(self, logger: Logger) -> None:
+        super().__init__()
+        self._logger = logger
+
+    @property
+    def example(self) -> Optional[str]:
+        return "$ protostar format"
+
+    @property
+    def name(self) -> str:
+        return "format"
+
+    @property
+    def description(self) -> str:
+        return "Format cairo sourcecode."
+
+    @property
+    def arguments(self) -> List[Command.Argument]:
+        return [
+            Command.Argument(
+                name="target",
+                description=("Directory to format."),
+                type="str",
+                is_array=True,
+                is_positional=True,
+                default=["."],
+            ),
```

Comment:
could this be an optional positional argument like clang-format or rustfmt have?

```
usage: rustfmt [options] <file>...
```

```
USAGE: clang-format [options] [<file> ...]
```

```
Usage: prettier [options] [file/dir/glob ...]
```